Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/PropertyEditorAPI_Internals.h

```diff
@@ -354,28 +366,79 @@ namespace AzToolsFramework
                 }
             }
 
-            auto value = AZ::DocumentPropertyEditor::Nodes::PropertyEditor::Value.ExtractFromDomNode(node);
-            if (value.has_value())
-            {
-                m_proxyValue = AZ::Dom::Utils::ValueToType<WrappedType>(value.value()).value_or(m_proxyValue);
-            }
-
             AZ::SerializeContext* serializeContext = nullptr;
             AZ::ComponentApplicationBus::BroadcastResult(serializeContext, &AZ::ComponentApplicationBus::Events::GetSerializeContext);
             AZ_Assert(serializeContext, "Serialization context not available");
 
-            // Set the m_genericClassInfo (if any) for our type in case the property handler needs to access it
-            // when downcasting to the generic type (e.g. for asset property downcasting to the generic AZ::Data::Asset<AZ::Data::AssetData>)
-            auto typeIdAttribute = node.FindMember(PropertyEditor::ValueType.GetName());
-            AZ::TypeId typeId = AZ::TypeId::CreateNull();
-            if (typeIdAttribute != node.MemberEnd())
+            AZ::TypeId typeId;
+            auto value = AZ::DocumentPropertyEditor::Nodes::PropertyEditor::Value.ExtractFromDomNode(node);
+            if (value.has_value())
             {
-                typeId = AZ::Dom::Utils::DomValueToTypeId(typeIdAttribute->second);
+                if (auto valueToType = AZ::Dom::Utils::ValueToType<WrappedType>(value.value()); valueToType)
+                {
+                    m_proxyValue = *valueToType;
+                    typeId = m_proxyClassData.m_typeId;
+                }
+                else if (auto valueToPointerObject = AZ::Dom::Utils::ValueToType<AZ::PointerObject>(value.value()); valueToPointerObject && valueToPointerObject->IsValid())
+                {
+                    if (valueToPointerObject->m_typeId == m_proxyClassData.m_typeId)
+                    {
+                        m_proxyValue = *reinterpret_cast<WrappedType*>(valueToPointerObject->m_address);
+                        typeId = valueToPointerObject->m_typeId;
+                    }
+                    else
+                    {
+                        const AZ::SerializeContext::ClassData* pointerClassData =
+                            serializeContext != nullptr ? serializeContext->FindClassData(valueToPointerObject->m_typeId) : nullptr;
+
+                        auto proxyClassData = serializeContext->FindClassData(m_proxyClassData.m_typeId);
+
+                        void* proxyAddress{};
+                        if constexpr (AZStd::is_pointer_v<WrappedType>)
+                        {
+                            proxyAddress = m_proxyValue;
+                        }
+                        else
+                        {
+                            proxyAddress = &m_proxyValue;
+                        }
```

Comment:
Can you remove this block as it will cause an unused variable compiler failure when using clang?

My change to fix this is still in PR: https://github.com/o3de/o3de/pull/16904

Reply:
yep, fixed :)